### PR TITLE
azureclient tidy up

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

File: cmd/createorupdate/createorupdate.go [hunk 1]
```diff
@@ -23,6 +24,17 @@ import (
...
 
+const (
+	// https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/api/application_list
+	// To list and patch AAD applications, this code needs to have the clientID
+	// of an application with the following permissions:
+	// API: Windows Azure Active Directory
+	//   Delegated permissions:
+	//      Sign in and read user profile
+	//      Access the directory as the signed-in user
+	clientID = "5935b8e2-3915-409c-bfb2-865b7a9291e0"
```
Comment (kwoodson) [inline cmd/createorupdate/createorupdate.go]: Whose client id does this correspond?

**Comment by kwoodson** [comment]
I like the removal of code so that part looks good.  Just curious about the hardcoded clientID.

**Comment by asalkeld** [comment]
/lgtm

all the e2e are failing, but not the fault of this PR

**Comment by kargakis** [comment]
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/857/pull-ci-openshift-openshift-azure-master-e2e-azure/1241/build-log.txt

The context deadline was exceeded, but in the artifacts everything seems to have come up
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/857/pull-ci-openshift-openshift-azure-master-e2e-azure/1241/artifacts/e2e-azure/nodes
https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_openshift-azure/857/pull-ci-openshift-openshift-azure-master-e2e-azure/1241/artifacts/e2e-azure/pods

26m for a VM to come up is a lot of time though, maybe an issue in australiasoutheast? @amanohar can you ptal at any debug logs for the `e2e-ycllqpbt` resource group?

File: cmd/createorupdate/createorupdate.go (hunk 1 above)
Comment (jim-minter) [reply cmd/createorupdate/createorupdate.go]: @kwoodson you were right ... today this came back and bit me ;-)

**Pull request merged**